docs/examples/go-ipfs-as-a-library/main.go

```diff
@@ -0,0 +1,337 @@
+	fmt.Println("Spawning node on a temporary repo")
+	ipfs, err := spawnEphemeral(ctx)
+	if err != nil {
+		panic(fmt.Errorf("failed to spawn ephemeral node: %s", err))
+	}
+
+	fmt.Println("IPFS node is running")
+
+	/// --- Part II: Adding a file and a directory to IPFS
+
+	fmt.Println("\n-- Adding and getting back files & directories --")
+
+	inputBasePath := "./example-folder/"
+	inputPathFile := inputBasePath + "ipfs.paper.draft3.pdf"
+	inputPathDirectory := inputBasePath + "test-dir"
+
+	someFile, err := getUnixfsNode(inputPathFile)
+	if err != nil {
+		panic(fmt.Errorf("Could not get File: %s", err))
+	}
+
+	cidFile, err := ipfs.Unixfs().Add(ctx, someFile)
+	if err != nil {
+		panic(fmt.Errorf("Could not add File: %s", err))
+	}
+
+	fmt.Printf("Added file to IPFS with CID %s\n", cidFile.String())
+
+	someDirectory, err := getUnixfsNode(inputPathDirectory)
+	if err != nil {
+		panic(fmt.Errorf("Could not get File: %s", err))
+	}
+
+	cidDirectory, err := ipfs.Unixfs().Add(ctx, someDirectory)
+	if err != nil {
+		panic(fmt.Errorf("Could not add Directory: %s", err))
+	}
+
+	fmt.Printf("Added directory to IPFS with CID %s\n", cidDirectory.String())
+
+	/// --- Part III: Getting the file and directory you added back
+
+	outputBasePath := "./example-folder/"
+	outputPathFile := outputBasePath + strings.Split(cidFile.String(), "/")[2]
+	outputPathDirectory := outputBasePath + strings.Split(cidDirectory.String(), "/")[2]
+
+	rootNodeFile, err := ipfs.Unixfs().Get(ctx, cidFile)
+	if err != nil {
+		panic(fmt.Errorf("Could not get file with CID: %s", err))
+	}
+
+	err = files.WriteTo(rootNodeFile, outputPathFile)
+	if err != nil {
+		panic(fmt.Errorf("Could not write out the fetched CID: %s", err))
+	}
+
+	fmt.Printf("Got file back from IPFS (IPFS path: %s) and wrote it to %s\n", cidFile.String(), outputPathFile)
+
+	rootNodeDirectory, err := ipfs.Unixfs().Get(ctx, cidDirectory)
+	if err != nil {
+		panic(fmt.Errorf("Could not get file with CID: %s", err))
+	}
+
+	err = files.WriteTo(rootNodeDirectory, outputPathDirectory)
+	if err != nil {
+		panic(fmt.Errorf("Could not write out the fetched CID: %s", err))
+	}
+
+	fmt.Printf("Got directory back from IPFS (IPFS path: %s) and wrote it to %s\n", cidDirectory.String(), outputPathDirectory)
+
+	/// --- Part IV: Getting a file from the IPFS Network
+
+	fmt.Println("\n-- Going to connect to a few nodes in the Network as bootstrappers --")
+
+	bootstrapNodes := []string{
+		// IPFS Cluster Pinning nodes
+		"/ip4/138.201.67.219/tcp/4001/ipfs/QmUd6zHcbkbcs7SMxwLs48qZVX3vpcM8errYS7xEczwRMA",
+		"/ip4/138.201.67.220/tcp/4001/ipfs/QmNSYxZAiJHeLdkBg38roksAR9So7Y5eojks1yjEcUtZ7i",
+		"/ip4/138.201.68.74/tcp/4001/ipfs/QmdnXwLrC8p1ueiq2Qya8joNvk3TVVDAut7PrikmZwubtR",
+		"/ip4/94.130.135.167/tcp/4001/ipfs/QmUEMvxS2e7iDrereVYc5SWPauXPyNwxcy9BXZrC1QTcHE",
+
+		// IPFS Bootstrapper nodes. Currently you can't connect to them directly because they don't yet run with 2048 bit keys.
```

Comment:
Probably best to just remove these. The peerIds for the old bootstrap nodes point to identities that have 1024 bit length keys; changing the key length would (i assume) change the peer Id, so there is no fix here, other than to deprecate the old ones, now the new ones are running.

Reply:
Got your point. At the same time, I expect that the Bootstrapper nodes will be the first that folks will try to dial to when they expand on this example. When that happens, they will have no clue and therefore, having this note here reduces the chance of them getting lost or giving up.

Reply:
Ok. I would find it misleading to suggest there is a "fix" tho. The good news is that the "new" dnsaddr bootstrappers are up now, so you could include those here https://github.com/ipfs/go-ipfs-config/blob/ed60afb72517463df516d7d6ea0a98fc07369024/bootstrap_peers.go#L22-L25

More generally, I'm not convinced we should be rolling out a breaking change where by ipfs node can no-longer connect to older nodes... but that is a conversation for elsewhere.

Reply:
All nodes created since 2015 use 2048 bit keys.

Reply:
Updated, working like a charm :)